src/rime/gear/simplifier.cc

```diff
@@ -250,7 +250,13 @@ bool Simplifier::Convert(const an<Candidate>& original,
         if (forms[i] == original->text()) {
           result->push_back(original);
         } else {
-          PushBack(original, result, forms[i]);
+          string simplified;
+          const bool success_converted = opencc_->ConvertText(original->text(), &simplified);
+          if (success_converted && original->text().compare(simplified) != 0) {
+            PushBack(original, result, simplified);
+          } else {
+            PushBack(original, result, forms[i]);
+          }
```

Comment:
This is redundantly unnecessary. A simple switch of the `if` and `else` branches would suffice.
code actually replaces starting at #L246 till #L266
```suggestion
    string simplified;
    success = opencc_->ConvertText(original->text(), &simplified);
    if (success) {
      PushBack(original, result, simplified);
    } else {
      vector<string> forms;
      success = opencc_->ConvertWord(original->text(), &forms);
      if (success) {
        for (size_t i = 0; i < forms.size(); ++i) {
          if (forms[i] == original->text()) {
            result->push_back(original);
          } else {
            PushBack(original, result, forms[i]);
          }
        }
```

Reply:
I have not tested your suggestion. Would it lose results if character can be converted into multiple ones? 

For example, 「才」 would be converted to 「才」 and 「纔」, but it will use `ConvertText` first, get result 「才」, and go to `success` branch. Then the result 「纔」 will lose.

Reply:
> 

fair point. I did a bit testing, and realize the problem is at the logic of the function. The **chained** conversion is never realized. It's not about one-to-one vs one-to-multi, but even the definition of `success` is wrong.

`ConvertWord` only converts at most once, which is based on the assumption that rime would convert from `zh-hant`, and it wouldn't need a second or third instance of (chained) conversion, _which is not true if one were to convert from `zh-hant` to `zh-hans-CN` / `zh-hans-SG` / `zh-hans-MY` (there are minor differences between standard variants, in addition to the concurrent use of `zh-hans-MY` and `zh-hant-MY`._ 

`ConvertText` wrongly treats A->B->A as failed conversion (e.g. `zh-hans` 才 -> `zh-hant` 纔 -> `zh-hant-TW` / `zh-hant-HK` 才. In this case, `ConvertWord` stops at `zh-hant` 纔, failing to convert it back to `zh-hant-TW` 才 or `zh-hant-HK` 才.